backend/azure/dtmb/dtmb.go

```diff
@@ -0,0 +1,531 @@
+	endpoint                  string
+	options                   *DTMBOptions
+	orchestrationQueue        utils.SyncQueue[dtmbprotos.ExecuteOrchestrationMessage]
+	activityQueue             utils.SyncQueue[dtmbprotos.ExecuteActivityMessage]
+	orchestrationHistoryCache utils.OrchestrationHistoryCache
+	connectWorkerClientStream chan *dtmbprotos.ConnectWorkerClientMessage
+	clientClient              dtmbprotos.TaskHubClientClient
+	workerClient              dtmbprotos.TaskHubWorkerClient
+	workerCancelFunc          context.CancelFunc
+}
+
+type DTMBOptions struct {
+	Endpoint string
+}
+
+func NewDTMBOptions(endpoint string) *DTMBOptions {
+	if endpoint != "" {
+		return &DTMBOptions{
+			Endpoint: endpoint,
+		}
+	} else {
+		return &DTMBOptions{
+			Endpoint: defaultEndpoint,
+		}
+	}
+}
+
+func NewDTMB(opts *DTMBOptions, logger backend.Logger) (*dtmb, error) {
+	be := &dtmb{
+		logger: logger,
+	}
+
+	if opts == nil {
+		opts = NewDTMBOptions(defaultEndpoint)
+	}
+	be.options = opts
+	be.endpoint = opts.Endpoint
+
+	// The following queues are used to store messages received from the server
+	be.orchestrationQueue = utils.NewSyncQueue[dtmbprotos.ExecuteOrchestrationMessage]()
+	be.activityQueue = utils.NewSyncQueue[dtmbprotos.ExecuteActivityMessage]()
+
+	be.orchestrationHistoryCache = utils.NewOrchestrationHistoryCache(nil) // TODO: make capacity configurable
+
+	ctx := context.Background()
+	connCtx, connCancel := context.WithTimeout(ctx, 15*time.Second) // TODO: make this a configurable timeout
+	conn, err := grpc.DialContext(connCtx, be.endpoint, grpc.WithBlock(), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	connCancel()
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to dtmb: %v", err)
+	}
+
+	be.clientClient = dtmbprotos.NewTaskHubClientClient(conn)
+	be.workerClient = dtmbprotos.NewTaskHubWorkerClient(conn)
+
+	return be, nil
+}
+
+// CreateTaskHub creates a new task hub for the current backend. Task hub creation must be idempotent.
+//
+// If the task hub for this backend already exists, an error of type [ErrTaskHubExists] is returned.
+func (d *dtmb) CreateTaskHub(ctx context.Context) error {
+	_, err := d.clientClient.Metadata(ctx, &dtmbprotos.MetadataRequest{})
+	if err != nil {
+		return backend.ErrTaskHubNotFound
+	}
+	return nil
+}
+
+// DeleteTaskHub deletes an existing task hub configured for the current backend. It's up to the backend
+// implementation to determine how the task hub data is deleted.
+//
+// If the task hub for this backend doesn't exist, an error of type [ErrTaskHubNotFound] is returned.
+func (d *dtmb) DeleteTaskHub(context.Context) error {
+	return fmt.Errorf("the TaskHub cannot be deleted via the API. Please delete the TaskHub manually")
+}
+
+// Start starts any background processing done by this backend.
+func (d *dtmb) Start(ctx context.Context, orchestrators *[]string, activities *[]string) error {
+	// TODO: is the context provided a background context?
```

Comment:
At the start of this method, add a check to make sure that the object hasn't already been started, or bad things™ will happen. For exmaple use an `atomic.Bool running` and do `if !running.CompareAndSwap(false, true) { error }`